interfaces/builtin/network_manager.go

```diff
@@ -73,6 +73,9 @@ network packet,
 # from netplan
 /run/NetworkManager/{,**} r,
 
+# Allow writing dhcp lease files outside of the snap
```

Comment:
The idea is that the lease files should be somewhere that an external program can find them.